### ISSUE TITLE
Test round-trip of BSync with Audit Template and SEED

### ISSUE THREAD

**Issue by macintoshpie** [SEED-platform/seed#2743]
## Context
SEED is improving integration with Audit Template Tool.

## Steps

1. Download example BSync files from ATT:
 (here's a zip of the files downloaded: 
[Archive.zip](https://github.com/SEED-platform/seed/files/6962468/Archive.zip)

 - https://buildingenergyscore.energy.gov/documents/example_NYC_audit_report_property_BS_v2.3_033121.xml
 - https://buildingenergyscore.energy.gov/documents/example_SF_audit_report_BS_v2.3_033121.xml

2. Do the remaining steps for each file.
3. Upload BSync file into audit template tool: https://buildingenergyscore.energy.gov/rp/buildings/upload
4. Export Building as BSync using the button "Download" > "XML (In-Progress)"
5. Import the downloaded file into SEED using "Import BuildingSync"
6. Document the following:
  - successfully imported?
  - Does it accomplish all requirements for SEED in this [spreadsheet](https://docs.google.com/spreadsheets/d/1yrvHA0X2rj6ckulvNiED9WadhiztnRv5Xf9BJtP-HHs/edit#gid=0)? Look at first 8 rows where column C is "SEED".
    - Here's a summary:
      - SEED should default to Proposed for implementation status of Measures
      - SEED should default to "True" for Recommendation of measures
      - SEED now reads AnnualPeakElectricityReduction
      - SEED now reads the following from Measures:
        - MeasureTotalFirstCost
        - MeasureInstallationCost
        - MeasureMaterialCost
        - MVCost
        - UsefulLife
      - SEED now supports BSync versions 2.1, 2.2, and 2.3

**Comment by macintoshpie** [comment]
## Testing Example NYC file

This file failed validation against the SEED use case on selection tool:
<img width="1440" alt="Screen Shot 2021-08-10 at 10 26 57 AM" src="https://user-images.githubusercontent.com/18518728/128885876-cc38aad1-6575-40e4-a620-4af8e4f709a2.png">
[See full message here](https://gist.github.com/macintoshpie/55b02d4f100ffcc901e594f66bc1d5e8)

### Comments
This seems to have failed b/c there are two `auc:Building`s in the file:
<img width="553" alt="Screen Shot 2021-08-10 at 10 36 10 AM" src="https://user-images.githubusercontent.com/18518728/128886540-8e338504-4452-42a8-850f-ea0f36c2d3a9.png">

SEED expects only one building to be present in a BuildingSync file. The specific failures were due to the [assertion of expecting exactly one](https://github.com/BuildingSync/use-cases/blob/bc633208c5d266e64f03b640e7e4a7694221d557/SEED/schematron/SEED-1.0.0.sch#L8) of those asserted elements (e.g. exactly one StreetAddress but there were two).

#### Solutions
We have a few options:
- basically change nothing, continue failing when trying to import files like this. We should probably update the error message from the use case though b/c it's misleading (it says the file is missing an element when really the issue is that there's not exactly one of the element). Maybe we should also add another assertion that there's exactly one auc:Building
- Try to update SEED's import process to allow multiple buildings. This is not trivial. The biggest implication is that we'd need really explicit linking of non-building elements such as Scenarios, TimeSeries data, etc to the building itself. I _think_ we treat everything as belonging to the building implicitly b/c of our one to one file to building assumption, but maybe we already only look at properly linked elements. Another implication is handling BuildingSync exports -- we edit the imported file when exporting, but with multiple buildings in a single file things get a little more complicated

**Comment by macintoshpie** [comment]
## Testing Example SF file

- The file passed the use-case validation [with some warnings](https://gist.github.com/macintoshpie/8c4a3eafbc922fe8764dd1202470c4c6)
- About 1/2 of SEED's predefined BSync mapping profile fields were found in the file.
<img width="1177" alt="Screen Shot 2021-08-10 at 10 51 32 AM" src="https://user-images.githubusercontent.com/18518728/128889346-db1a2fa9-7615-4281-853d-11b7cad988a4.png">
<img width="1155" alt="Screen Shot 2021-08-10 at 10 51 41 AM" src="https://user-images.githubusercontent.com/18518728/128889371-5e126cf2-7de3-4c18-a843-2fa4f5dc1c65.png">
- when finalizing the mapping, SEED presented some warnings about skipped scenarios:

```
Warning(s)/Error(s) occurred while processing the file(s):
{
  "example_SF_audit_report_BS_v2.3_033121.xml": {
    "warnings": [
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975101191220",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975101670140",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975102339440",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975102677160",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975109355600",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975109429720",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975110442060",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975115564480",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975115697520",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975116031200",
      "Skipping scenario because it does not have a name. ID = ScenarioType-69975117102320"
    ],
    "errors": []
  }
}
```
Below are screenshots of these scenarios we are skipping:
ScenarioType-69975101191220
<img width="720" alt="Screen Shot 2021-08-10 at 10 56 15 AM" src="https://user-images.githubusercontent.com/18518728/128890061-fa986149-5831-4d10-bd82-d369daa0d931.png">

ScenarioType-69975101670140
<img width="540" alt="Screen Shot 2021-08-10 at 10 57 19 AM" src="https://user-images.githubusercontent.com/18518728/128890158-15b6eec0-24a1-4980-8189-6ac02bd3dcd1.png">

ScenarioType-69975102339440
<img width="544" alt="Screen Shot 2021-08-10 at 10 57 45 AM" src="https://user-images.githubusercontent.com/18518728/128890243-93b493a2-612e-4c06-b916-8a1cb42b70e7.png">

ScenarioType-69975102677160
<img width="534" alt="Screen Shot 2021-08-10 at 10 58 11 AM" src="https://user-images.githubusercontent.com/18518728/128890324-d6a7a529-b9a2-4c8d-af34-373d61b41b77.png">

ScenarioType-69975109355600
<img width="535" alt="Screen Shot 2021-08-10 at 10 58 49 AM" src="https://user-images.githubusercontent.com/18518728/128890443-a5283028-6441-4459-9339-868e5fede5dc.png">

ScenarioType-69975109429720
<img width="529" alt="Screen Shot 2021-08-10 at 10 59 23 AM" src="https://user-images.githubusercontent.com/18518728/128890551-57a1e709-1822-4ad2-bb2c-de2ecfc8edac.png">

ScenarioType-69975110442060
<img width="532" alt="Screen Shot 2021-08-10 at 11 00 00 AM" src="https://user-images.githubusercontent.com/18518728/128890679-d722739b-1fdf-4e91-adad-f69c385d52cf.png">

ScenarioType-69975115564480
<img width="543" alt="Screen Shot 2021-08-10 at 11 00 38 AM" src="https://user-images.githubusercontent.com/18518728/128890834-7b610969-07e1-4004-9169-624485ad1cff.png">

ScenarioType-69975115697520
<img width="530" alt="Screen Shot 2021-08-10 at 11 02 33 AM" src="https://user-images.githubusercontent.com/18518728/128891088-ab5f5388-bfc7-495f-af75-297136dcfb79.png">

ScenarioType-69975116031200
<img width="476" alt="Screen Shot 2021-08-10 at 11 03 04 AM" src="https://user-images.githubusercontent.com/18518728/128891172-59362db8-1e0c-4719-ba6c-a3e6da0c7b0f.png">

ScenarioType-69975117102320
<img width="535" alt="Screen Shot 2021-08-10 at 11 03 41 AM" src="https://user-images.githubusercontent.com/18518728/128891261-e935dc60-35ff-4aa0-bf6e-ceea86f7688a.png">


Due to the scenarios being skipped, no meter data was imported for the property
<img width="1434" alt="Screen Shot 2021-08-10 at 11 04 31 AM" src="https://user-images.githubusercontent.com/18518728/128891482-b1574aa0-3275-4070-ad1c-0467884d6216.png">

All measures measures (2) and all package of measures (ie scenarios) (2)  were successfully imported:
<img width="1379" alt="Screen Shot 2021-08-10 at 11 05 36 AM" src="https://user-images.githubusercontent.com/18518728/128891601-bb1abdac-742f-4a5c-9059-332ae1f382d6.png">

<img width="1378" alt="Screen Shot 2021-08-10 at 11 05 44 AM" src="https://user-images.githubusercontent.com/18518728/128891626-e3ef4c72-a3f4-4284-ae8c-e8b24175029e.png">

### Comments
- We have discussed with AT folks the need for scenarios to have names, and I believe they have updated this in their codebase but maybe not in the example files. I will get a more updated example and test this again.
- Other than the skipped scenarios, things seem to be working well.

**Comment by macintoshpie** [comment]
## Testing example SF file with Scenario Names
⚠️ NOTE that this was tested on a version of SEED not currently merged into develop: b1a956f92f0b0b55248c42543636f4d96b4e1c9a
Test file: 
[example_SF_audit_report_BS_v2.3_033121_with_names.xml.zip](https://github.com/SEED-platform/seed/files/6976921/example_SF_audit_report_BS_v2.3_033121_with_names.xml.zip)

- the file passed use case validation, same as original SF file
- SEED no longer skips scenarios b/c the file has scenario names. It does skip meters and meter readings though:

<img width="615" alt="Screen Shot 2021-08-12 at 12 15 41 PM" src="https://user-images.githubusercontent.com/18518728/129231529-48f24cdd-b530-4364-9321-01f708ca6569.png">

Full warnings:
```
Warning(s)/Error(s) occurred while processing the file(s):
{
  "buildings_xml_report (12) (1).xml": {
    "warnings": [
      "Skipping resource use ResourceUseType-70066244180560 due to missing type or units",
      "Skipped meter ResourceUseType-70066237266060 because it had no valid readings",
      "Skipped meter ResourceUseType-70066237389520 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242363280 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242542420 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242602540 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242676280 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242709380 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242791660 because it had no valid readings",
      "Skipped meter ResourceUseType-70066242835640 because it had no valid readings"
    ],
    "errors": []
  }
}
```

- There are some "empty" scenarios after importing. We are going to discuss how we want to handle this soon, but I am guessing this is probably OK for now

<img width="1440" alt="Screen Shot 2021-08-12 at 12 25 28 PM" src="https://user-images.githubusercontent.com/18518728/129235711-24953d92-2c8d-487f-8d65-c8a40396e3cb.png">

### Comments
- I believe this meets our requirements originally laid out
- There is still additional work we could do to improve the integration between the applications (e.g. empty scenarios)

**Comment by macintoshpie** [comment]
## Test example SF file with scenario names
⚠️ NOTE that this was tested on a version of SEED not currently merged into develop: b1a956f92f0b0b55248c42543636f4d96b4e1c9a
Test file -- I edited the original SF example file linked above by adding scenario names manually:
[example_SF_audit_report_BS_v2.3_033121_with_names.xml.zip](https://github.com/SEED-platform/seed/files/6977171/example_SF_audit_report_BS_v2.3_033121_with_names.xml.zip)

- Same comments as original SF file, but this time it properly imported meter data b/c the scenario now has a name
<img width="1440" alt="Screen Shot 2021-08-12 at 12 56 55 PM" src="https://user-images.githubusercontent.com/18518728/129237454-acb39512-6e6b-4976-85a9-8332f478d8e7.png">

- Same comments as ATL file about "empty" scenarios

**Comment by macintoshpie** [comment]
After discussing this we decided:
- we only want to create SEED Scenarios if the BSync Scenario is a package of measures. But make sure 
- still create meter readings for these package of measure scenarios _AND_ for non-POM scenarios
